### PR TITLE
Feat/added edit test flow name

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

#1443 
In this PRi have added the functionality for edit the name of the test flow , one thing is remaining that will be done when we are having the  save test flow feature

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts [hunk 1]
```diff
@@ -542,4 +545,27 @@ export class TestflowExplorerPageViewModel {
...
+   */
+    public updateName = async (_name: any, event = "") => {
+      const progressiveTab = createDeepCopy(this._tab.getValue());
+      if (event === "blur" && _name === "") {
+        const data = await this.testflowRepository.readTestFlow(
+          progressiveTab.id,
+        );
+        progressiveTab.name = data.name;
+      } else if (event === "") {
+        progressiveTab.name = _name;
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts:563]: update comment environment should be replaced with testflow

File: src/packages/@app/repositories/testflow.repository.ts [hunk 2]
```diff
@@ -93,6 +93,17 @@ export class TestflowRepository {
       .exec();
   };
 
+
+  public readTestFlow = async (uuid: string): Promise<any> => {
+    return await RxDB.getInstance()
+      .rxdb.environment.findOne({
+        selector: {
+          id: uuid,
+        },
```
Comment (itsmdasifraza) [inline src/packages/@app/repositories/testflow.repository.ts:102]: rxdb.testflow

File: src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte [hunk 3]
```diff
@@ -637,4 +658,35 @@
...
+
+  .search-input-container:hover {
+    border: 1px solid var(--border-primary-300);
+    caret-color: var(--border-primary-300);
+  }
+  .search-input-container:focus-within {
+    border-color: var(--border-primary-300);
+    caret-color: var(--border-primary-300);
+  }
 </style>
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte:692]: use bootstrap class

File: src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte [hunk 4]
```diff
@@ -426,17 +426,38 @@
...
+          class={`bg-transparent w-100 border-0 my-auto`}
+          on:input={(e) => {
+            testFlowName = e.target.value;
+            handleTestFlowNameChange(testFlowName, "");
+          }}
+          on:blur={(e) => {
+            handleTestFlowNameChange(testFlowName, "blur");
+          }}
+          value={testFlowName}
+        />
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte:459]: push this to a component

File: src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts [hunk 5]
```diff
@@ -542,4 +545,27 @@ export class TestflowExplorerPageViewModel {
...
+
+
+    /**
+   * @description - updates testflow tab name
+   * @param _name - new test flow name
+   */
+    public updateName = async (_name: any, event = "") => {
+      const progressiveTab = createDeepCopy(this._tab.getValue());
+      if (event === "blur" && _name === "") {
+        const data = await this.testflowRepository.readTestFlow(
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts:558]: data type name should be string

File: src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts (hunk 1 above)
Comment (Abhishek9503) [reply src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts]: Resovled

File: src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte (hunk 3 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte]: Resolved

File: src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts (hunk 5 above)
Comment (Abhishek9503) [reply src/packages/@app/pages/TestflowExplorerPage/TestflowExplorerPage.ViewModel.ts:558]: Resolved

File: src/packages/@app/repositories/testflow.repository.ts (hunk 2 above)
Comment (Abhishek9503) [reply src/packages/@app/repositories/testflow.repository.ts]: Resolved

File: src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte (hunk 4 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/features/testflow-explorer/layout/TestflowExplorer.svelte:459]: Resolved

**Review by itsmdasifraza — APPROVED** [review]

**Pull request merged**